include/numerics/petsc_matrix.h

```diff
@@ -163,8 +168,25 @@ class PetscMatrix final : public SparseMatrix<T>
 
   virtual numeric_index_type m () const override;
 
+  /**
+   * Get the number of rows owned by this process
+   */
+  numeric_index_type local_m () const;
+
   virtual numeric_index_type n () const override;
 
+  /**
+   * Get the number of columns owned by this process
+   */
+  numeric_index_type local_n () const;
```

Comment:
I'm confused by this one.  My first interpretation here is "the total number of columns with non-empty entries in the sparsity pattern in rows owned by this process", but I'm not actually sure how PETSc would determine that - we give them n_nz and n_oz rather than complete sparsity info, right?  And without the latter there's no way to immediately determine which non-empty columns from which rows overlap?  So is this method only well-defined for matrices that have already been assembled??

Reply:
It is not related to nnz, noz or matrix assembly. It is a layout thing. It is something we tell PETSc to use in libMesh.

In `PetscMatrix::init()`, we have something like 

```
ierr = MatSetSizes(_mat, m_local, n_local, m_global, n_global);
  LIBMESH_CHKERR(ierr);
```

Reply:
`"n_local"` is `"local_n()"`

Reply:
Thanks @fdkong. This is what I had determined through experimentation this morning :smile:

Reply:
That certainly clears it up!  Thanks!